src/static/controllers/vizualizationController.js

```diff
@@ -137,25 +139,54 @@ export default class VizualizationController {
+        let projection = {}
+        cur_join_keys.map(key => projection[key] = [key, 'IDENTITY'])
+        let joinTable = await this.getData(tablename, [],projection,null, cur_join_keys, 1000, selected_join_values)
 
         let links1 = this.getEdges(tablename)
         let links2 = this.getEdges(next_tablename)
-        
-        // this.visView.drawSingleTable(tablename, data1["header"], data1["data"], links1, null, visDiv, this.exploreHandler)
-        this.visView.drawSingleTable(next_tablename, data2["header"], data2["data"], links2, null, element, this.exploreHandler)
+
+        // generate cell height array by counting the number of rows in left table with the same join key
+        let cellHeights = []
+        let leftcellHeights = []
+        let rightCellHeights = []
+        let cur_join_key_idxs = cur_join_keys.map( key => leftTableData["header"].indexOf(key))
+        let cur_join_key_tuples = leftTableData["data"].map(row => cur_join_key_idxs.map(idx => row[idx]))
+        let cur_join_key_set = new Set(cur_join_key_tuples.map(JSON.stringify))
+        // get right table data but filter for only join key values
+        let next_selection_conds = Array.from(cur_join_key_set).map(JSON.parse).map(tuple => tuple.map((key, idx) => next_join_keys[idx] + " = " + key).join(" AND ")).join(" OR ")
+        let rightTableData = await this.getData(next_tablename, [next_selection_conds],null,null, next_join_keys, 1000, selected_join_values)
+
+        let next_join_key_idxs = next_join_keys.map(key => rightTableData["header"].indexOf(key))
+        Array.from(cur_join_key_set).map(JSON.parse).forEach(key_tuple => {
+            let l_count = cur_join_key_tuples.filter(k => JSON.stringify(k) === JSON.stringify(key_tuple)).length
+            let r_count = rightTableData["data"].filter(row => JSON.stringify(next_join_key_idxs.map(idx => row[idx])) === JSON.stringify(key_tuple)).length
+            let max_count = Math.max(l_count, r_count)
+            cellHeights.push(max_count)
+            // push to array l_count times
+            leftcellHeights.push(...Array(l_count).fill(max_count/l_count))
+            rightCellHeights.push(...Array(r_count).fill(max_count/r_count))
+        });
+
+
+        // grey out all existing tables in visView
+        this.visView.greyOutAllTables(tableIdx);
+        // delete all tables after the current table (including the current table) so we can redraw them
+        this.visView.deleteTablesAfter(tableIdx);
+        this.visView.drawSingleTable(tablename, leftTableData["header"], leftTableData["data"], links1, leftcellHeights, null, this.exploreHandler)
+        this.visView.drawSingleTable("-", joinTable["header"], Array.from(cur_join_key_set).map(JSON.parse), [], cellHeights, null, this.exploreHandler)
+        this.visView.drawSingleTable(next_tablename, rightTableData["header"], rightTableData["data"], links2, rightCellHeights, null, this.exploreHandler)
```

Comment:
Look cool. I haven't delved into details. Will make a pass on this part when the codes can work.